In [2]:
import redis

r = redis.StrictRedis(
    host="c-c9qs6ld6nntbvr2u5o6e.rw.mdb.yandexcloud.net",
    port=6380,
    password="de_password",
    ssl=True,
    ssl_ca_certs="/home/tim/.redis/YandexInternalRootCA.crt",
)

r.set("foo", "bar")
print(r.get("foo"))

b'bar'


In [3]:
import redis

def main():
# указываем параметры, которые необходимы для подключения
    host="c-c9qs6ld6nntbvr2u5o6e.rw.mdb.yandexcloud.net"
    port=6380
    password="de_password"
    ca_path="/home/tim/.redis/YandexInternalRootCA.crt"

# инициализируем клиент, с помощью которого будем подключаться к Redis
    client = redis.StrictRedis(
            host=host,
            port=port,
            password=password,
            ssl=True,
            ssl_ca_certs=ca_path)

# просто подключиться недостаточно, да и неинтересно: нужно же ещё и поработать с хранилищем

# запишем в Redis ключ here_is_the_key и соответствующее ему строковое значение string value
    client.set("here_is_the_key", "string value")

# получим значение ключа с помощью команды get
    result = client.get("here_is_the_key")
    if not result:
        print("Запись с указанным ключом не найдена.")
        return

    result = result.decode("utf-8")
    print(result)

if __name__ == '__main__':
    main()

string value


In [4]:
import redis

def main():
# указываем параметры, которые необходимы для подключения
    host="c-c9qs6ld6nntbvr2u5o6e.rw.mdb.yandexcloud.net"
    port=6380
    password="de_password"
    ca_path="/home/tim/.redis/YandexInternalRootCA.crt"

# инициализируем клиент, с помощью которого будем подключаться к Redis
    client = redis.StrictRedis(
            host=host,
            port=port,
            password=password,
            ssl=True,
            ssl_ca_certs=ca_path)

# просто подключиться недостаточно, да и неинтересно: нужно же ещё и поработать с хранилищем


# получим значение ключа с помощью команды get
    result = client.get("a51e4e31ae4602047ec52534")
    if not result:
        print("Запись с указанным ключом не найдена.")
        return

    result = result.decode("utf-8")
    print(result)

if __name__ == '__main__':
    main()

{"_id": "a51e4e31ae4602047ec52534", "name": "Кубдари", "menu": [{"_id": "3de9442da57a50ccfaf72866", "name": "Пхали с фасолью", "price": 200, "category": "Закуски"}, {"_id": "c1b046c8b2184d586ea74928", "name": "Ассорти пхали", "price": 290, "category": "Закуски"}, {"_id": "d9e3403e806ad914f08ab416", "name": "Долма", "price": 400, "category": "Закуски"}, {"_id": "6f8c430e9883b87eb3849dc0", "name": "Бадриджани с орехами", "price": 400, "category": "Закуски"}, {"_id": "c5b64c8ba71adb08ce4bf7bc", "name": "Аджапсандали", "price": 400, "category": "Закуски"}, {"_id": "165b4812b4334fdbeaeca3a3", "name": "Лобио", "price": 350, "category": "Закуски"}, {"_id": "4a8b4c698fc9d21622d9ff25", "name": "Сациви из курицы", "price": 350, "category": "Закуски"}, {"_id": "3dc44b0684e19baecc9d19e1", "name": "Пхали со шпинатом", "price": 200, "category": "Закуски"}, {"_id": "844c4628a01217273b8ac0d7", "name": "Пхали со свеклой", "price": 200, "category": "Закуски"}, {"_id": "a56c4b70b7d0e6522ccb3c9f", "name":

In [7]:
import redis

def main():
# указываем параметры, которые необходимы для подключения
    host="c-c9qs6ld6nntbvr2u5o6e.rw.mdb.yandexcloud.net"
    port=6380
    password="de_password"
    ca_path="/home/tim/.redis/YandexInternalRootCA.crt"

# инициализируем клиент, с помощью которого будем подключаться к Redis
    client = redis.StrictRedis(
            host=host,
            port=port,
            password=password,
            ssl=True,
            ssl_ca_certs=ca_path)

# просто подключиться недостаточно, да и неинтересно: нужно же ещё и поработать с хранилищем


# получим значение ключа с помощью команды get
    result = client.get("ef8c42c19b7518a9aebec106")
    if not result:
        print("Запись с указанным ключом не найдена.")
        return

    result = result.decode("utf-8")


    print(result)

if __name__ == '__main__':
    main()

{"_id": "ef8c42c19b7518a9aebec106", "name": "Вкус Индии", "menu": [{"_id": "22744fcdb947be9aa795e797", "name": "Бомбей Тикки", "price": 290, "category": "Закуски"}, {"_id": "1878439998067c1905f455b3", "name": "Чиккен Тикка", "price": 590, "category": "Закуски"}, {"_id": "95f74625be0428e4b2380310", "name": "Проун Пакора", "price": 790, "category": "Закуски"}, {"_id": "55084ea193269b89bc1686c4", "name": "Машрум Тикка", "price": 390, "category": "Закуски"}, {"_id": "e86d440f87ba45c8280310dc", "name": "Чили Проун", "price": 790, "category": "Закуски"}, {"_id": "927b4ba0b5494778ba576fd3", "name": "Микс Ведж Пакора", "price": 450, "category": "Закуски"}, {"_id": "", "name": "", "price": 0, "category": ""}, {"_id": "413c484db69df84af89cab35", "name": "Чили Машрум", "price": 399, "category": "Закуски"}, {"_id": "6f30486386ee88e590de632b", "name": "Панир Тикка", "price": 590, "category": "Закуски"}, {"_id": "1bda42a3819f598a052d2966", "name": "Чили Чикен", "price": 499, "category": "Закуски"}, 

b'bar'
string value
{"_id": "a51e4e31ae4602047ec52534", "name": "Кубдари", "menu": [{"_id": "3de9442da57a50ccfaf72866", "name": "Пхали с фасолью", "price": 200, "category": "Закуски"}, {"_id": "c1b046c8b2184d586ea74928", "name": "Ассорти пхали", "price": 290, "category": "Закуски"}, {"_id": "d9e3403e806ad914f08ab416", "name": "Долма", "price": 400, "category": "Закуски"}, {"_id": "6f8c430e9883b87eb3849dc0", "name": "Бадриджани с орехами", "price": 400, "category": "Закуски"}, {"_id": "c5b64c8ba71adb08ce4bf7bc", "name": "Аджапсандали", "price": 400, "category": "Закуски"}, {"_id": "165b4812b4334fdbeaeca3a3", "name": "Лобио", "price": 350, "category": "Закуски"}, {"_id": "4a8b4c698fc9d21622d9ff25", "name": "Сациви из курицы", "price": 350, "category": "Закуски"}, {"_id": "3dc44b0684e19baecc9d19e1", "name": "Пхали со шпинатом", "price": 200, "category": "Закуски"}, {"_id": "844c4628a01217273b8ac0d7", "name": "Пхали со свеклой", "price": 200, "category": "Закуски"}, {"_id": "a56c4b70b7d0e6522ccb3c9f", "name": "Салат Греческий", "price": 350, "category": "Салаты"}, {"_id": "96ca4cf8b0217ce521be2389", "name": "Салат Тбилисо", "price": 450, "category": "Салаты"}, {"_id": "dac24794ac36b29e09e8caf0", "name": "Салаи со сладкими помидорами", "price": 350, "category": "Салаты"}, {"_id": "41ff432396c5c7c414e5a4a9", "name": "Сала по-грузински с орехом", "price": 350, "category": "Салаты"}, {"_id": "dcfd472aa0f182837473d42d", "name": "Куриный суп с лапшой", "price": 300, "category": "Супы"}, {"_id": "b18542b1901ef95b547bb1f3", "name": "Харчо из говядины", "price": 300, "category": "Супы"}, {"_id": "22e74a698d81d8b5275e2ba7", "name": "Борщ", "price": 300, "category": "Супы"}, {"_id": "54cb4508a729850ecd590ac0", "name": "Хачапургер с говядиной", "price": 450, "category": "Основные блюда"}, {"_id": "1c3744b29df6178f396fb4bd", "name": "Хинкали с бараниной 3 шт", "price": 270, "category": "Основные блюда"}, {"_id": "564b4a2bb6e699b15549069c", "name": "Хинкали с говядиной и свининой 3 шт", "price": 270, "category": "Основные блюда"}, {"_id": "a8d54fb4abc3d87722d9ea26", "name": "Хинкали с сыром сулугуни 3 шт", "price": 240, "category": "Основные блюда"}, {"_id": "845d42f8af7705492d2d576a", "name": "Хинкали с креветкой Том Ям 3 шт", "price": 450, "category": "Основные блюда"}, {"_id": "c6d04ae191fc4813b277fb1d", "name": "Хинкали с курицей и зеленью 3 шт", "price": 240, "category": "Основные блюда"}, {"_id": "1f22499787cfda2be82a71e7", "name": "Оджахури с грибами и зеленью", "price": 400, "category": "Основные блюда"}, {"_id": "562647ee8273611413325523", "name": "Чкмерули с курицей", "price": 460, "category": "Основные блюда"}, {"_id": "84544c10b6a1de9310cd0462", "name": "Оджахури с говядиной", "price": 560, "category": "Основные блюда"}, {"_id": "c85c4962ac86a2708cba34ad", "name": "Чахохбили", "price": 420, "category": "Основные блюда"}, {"_id": "76ee4f718d3309dcf01914eb", "name": "Чашушули", "price": 460, "category": "Основные блюда"}, {"_id": "346a499da9d4d9c81ce87f7a", "name": "Оджахури со свининой", "price": 550, "category": "Основные блюда"}, {"_id": "b9cb4b79bcc662db0be0844e", "name": "Хинкали с вишней 3 шт", "price": 240, "category": "Основные блюда"}, {"_id": "0ff94c80a0816057090ed20c", "name": "Пахлава", "price": 290, "category": "Десерты"}, {"_id": "09d24a25ae51f46def8e5965", "name": "Лобиани лодочка", "price": 350, "category": "Выпечка"}, {"_id": "129a492c90582164bc69f593", "name": "Кубдари с курицей и сыром", "price": 450, "category": "Выпечка"}, {"_id": "04e14e08b9fe5662c2414ae7", "name": "Мини кубдари с бараниной", "price": 250, "category": "Выпечка"}, {"_id": "083e482986816c6ab32d69d3", "name": "Кубдари с бараниной", "price": 450, "category": "Выпечка"}, {"_id": "bf65470ca47ae56177951569", "name": "Кубдари с говядиной с свининой", "price": 450, "category": "Выпечка"}, {"_id": "e4b5447fa889192c3aa97ec1", "name": "Хачапури со шпинатом и сыром", "price": 450, "category": "Выпечка"}, {"_id": "30fc434da23b451a314c323f", "name": "Хачапури по-аджарски с томатами", "price": 430, "category": "Выпечка"}, {"_id": "79594852878ed8bb3e093a57", "name": "Хачапури по-мегрельски", "price": 390, "category": "Выпечка"}, {"_id": "6af740d0b2b49d74de0bec85", "name": "Хачапури по-аджарски", "price": 390, "category": "Выпечка"}, {"_id": "1acb452ebcc8c462953a121d", "name": "Хачапури с картофелем и грибами", "price": 350, "category": "Выпечка"}, {"_id": "60fd41ab8c8f919e7ee17634", "name": "Хачапури по-имеретински", "price": 390, "category": "Выпечка"}, {"_id": "c92f41deb2e95a4187aad796", "name": "Грузинский хлеб шоти", "price": 50, "category": "Выпечка"}, {"_id": "a46c44c7a2116150664eec02", "name": "Мини кубдари с курицей и сыром", "price": 250, "category": "Выпечка"}, {"_id": "1b3e42939c414bc1ea69e2fe", "name": "Мини кубдари с говядиной и свининой", "price": 250, "category": "Выпечка"}, {"_id": "60b4455ead3234b9575b8b66", "name": "Лимонад грузинский", "price": 190, "category": "Напитки"}, {"_id": "41cd40de86c2728eb72b819d", "name": "Вода Свирме", "price": 190, "category": "Напитки"}, {"_id": "2d614774b2d57fd0650a3129", "name": "Соус Сметана", "price": 50, "category": "Соусы"}, {"_id": "28d949a99671154acd0d0160", "name": "Соус Сметана с чесноком", "price": 50, "category": "Соусы"}, {"_id": "48ae4b6791696398de59d8f7", "name": "Соус Сацибели", "price": 50, "category": "Соусы"}, {"_id": "c97b4da89832c190235d5961", "name": "Соус Аджика", "price": 50, "category": "Соусы"}, {"_id": "45c34a7eaea04e757faee967", "name": "Соус Наршараб", "price": 100, "category": "Соусы"}, {"_id": "24094fbba9e3187d52e0af77", "name": "Соус Ткемали", "price": 50, "category": "Соусы"}], "update_ts_utc": "2023-04-15 08:19:21"}
{"_id": "ef8c42c19b7518a9aebec106", "name": "Вкус Индии", "menu": [{"_id": "22744fcdb947be9aa795e797", "name": "Бомбей Тикки", "price": 290, "category": "Закуски"}, {"_id": "1878439998067c1905f455b3", "name": "Чиккен Тикка", "price": 590, "category": "Закуски"}, {"_id": "95f74625be0428e4b2380310", "name": "Проун Пакора", "price": 790, "category": "Закуски"}, {"_id": "55084ea193269b89bc1686c4", "name": "Машрум Тикка", "price": 390, "category": "Закуски"}, {"_id": "e86d440f87ba45c8280310dc", "name": "Чили Проун", "price": 790, "category": "Закуски"}, {"_id": "927b4ba0b5494778ba576fd3", "name": "Микс Ведж Пакора", "price": 450, "category": "Закуски"}, {"_id": "", "name": "", "price": 0, "category": ""}, {"_id": "413c484db69df84af89cab35", "name": "Чили Машрум", "price": 399, "category": "Закуски"}, {"_id": "6f30486386ee88e590de632b", "name": "Панир Тикка", "price": 590, "category": "Закуски"}, {"_id": "1bda42a3819f598a052d2966", "name": "Чили Чикен", "price": 499, "category": "Закуски"}, {"_id": "6bb54650b9fef09054c545ef", "name": "Чили Панир", "price": 450, "category": "Закуски"}, {"_id": "1c6d4a019b936d6209c3fbe8", "name": "Чикен Сиксти Файв", "price": 490, "category": "Закуски"}, {"_id": "dd234386b138dfd863c7c7c5", "name": "Луковые Бхаджи", "price": 290, "category": "Закуски"}, {"_id": "e9214c6eab68e7625e6948c8", "name": "Панир Пакора", "price": 449, "category": "Закуски"}, {"_id": "cf0e4699b6ab70ad3df5d162", "name": "Мург Малай Тикка", "price": 550, "category": "Закуски"}, {"_id": "c0d44e02bbe3fa8bf7fcdf85", "name": "Чикен Шорба", "price": 380, "category": "Супы"}, {"_id": "ca8f449eadf0832b800fa62b", "name": "Овощной суп", "price": 349, "category": "Супы"}, {"_id": "b551423ca4daabd9e219ee47", "name": "Томатный суп", "price": 299, "category": "Супы"}, {"_id": "353d4799b8cb9891c9f32aeb", "name": "Суп с креветками", "price": 599, "category": "Супы"}, {"_id": "35dd4493974018de1ed0e036", "name": "Рис с шафраном", "price": 200, "category": "Гарниры"}, {"_id": "54ff4abd913cfd8a0e5b75b0", "name": "Простой рис Басмати", "price": 120, "category": "Гарниры"}, {"_id": "b8014b42b9e36df9befeffd3", "name": "Рис с кумином", "price": 175, "category": "Гарниры"}, {"_id": "aeca4d08abab78869c20128c", "name": "Рис с лимоном", "price": 175, "category": "Гарниры"}, {"_id": "32ed48df80469389087c3e53", "name": "Палак Панир", "price": 499, "category": "Основные блюда"}, {"_id": "dadb4f10954de26867b687b0", "name": "Машрум Маттар", "price": 450, "category": "Основные блюда"}, {"_id": "7bbd451a9ccb03e2f52badc9", "name": "Фиш Карри", "price": 595, "category": "Основные блюда"}, {"_id": "62a9453ca09d774bcea1ad45", "name": "Бриани с ягненком", "price": 650, "category": "Основные блюда"}, {"_id": "fa9e41cab58e5c525a74c326", "name": "Муттон Роган Джош", "price": 749, "category": "Основные блюда"}, {"_id": "080d4248a4673012342b116a", "name": "Муттон Карри", "price": 649, "category": "Основные блюда"}, {"_id": "800d4b1283e01fb243124225", "name": "Бриани с овощами", "price": 499, "category": "Основные блюда"}, {"_id": "7b26444b845e2740c8669b46", "name": "Чикке Карри", "price": 499, "category": "Основные блюда"}, {"_id": "3f7848ea94b146dedf030c2a", "name": "Дал Макхни", "price": 499, "category": "Основные блюда"}, {"_id": "e2cf449a9a96e33881fe50a5", "name": "Дал Тарка", "price": 390, "category": "Основные блюда"}, {"_id": "36d54f529ef3ac5e05210178", "name": "Баттер Чикке", "price": 599, "category": "Основные блюда"}, {"_id": "aab045a1b82908e59d225e9a", "name": "Бриани с курицей", "price": 550, "category": "Основные блюда"}, {"_id": "760244ac855a44c5287cb2ac", "name": "Панир Баттер Масала", "price": 550, "category": "Основные блюда"}, {"_id": "523f456e836a4c64203ef81e", "name": "Чиккен Тикка Масала", "price": 649, "category": "Основные блюда"}, {"_id": "9df848fe96dd0b097380d456", "name": "Бриани с креветками", "price": 690, "category": "Основные блюда"}, {"_id": "48b94736be0c02a1b84da6b4", "name": "Чанна Масала", "price": 390, "category": "Основные блюда"}, {"_id": "9228453584d3056edaca3e90", "name": "Чиз Наан", "price": 225, "category": "Выпечка"}, {"_id": "99654af1bbb100533109759e", "name": "Чиз Гарлик Наан", "price": 250, "category": "Выпечка"}, {"_id": "a2f74e5090c92413957b318b", "name": "Амритсари Кулча", "price": 150, "category": "Выпечка"}, {"_id": "47b947298daea892c7ae6c5f", "name": "Баттер Наан", "price": 120, "category": "Выпечка"}, {"_id": "0c92483792353f5e2773d75a", "name": "Гарлик Наан", "price": 150, "category": "Выпечка"}, {"_id": "ba57488684df2dd926883ca7", "name": "Алу Кулча", "price": 150, "category": "Выпечка"}, {"_id": "2160406ba7c026fd829ea738", "name": "Чай Масала", "price": 150, "category": "Чай"}, {"_id": "2c0b49dd92d3e411f8f5fabd", "name": "Сладкий Ласси", "price": 150, "category": "Напитки"}, {"_id": "7105441f860b1124a3a24fe9", "name": "Банана Ласси", "price": 250, "category": "Напитки"}, {"_id": "aec44b7b9da0e68d7b850fea", "name": "Джира Ласси", "price": 150, "category": "Напитки"}], "update_ts_utc": "2023-04-15 08:19:21"}

In [8]:
import redis

def main():
# указываем параметры, которые необходимы для подключения
    host="c-c9qs6ld6nntbvr2u5o6e.rw.mdb.yandexcloud.net"
    port=6380
    password="de_password"
    ca_path="/home/tim/.redis/YandexInternalRootCA.crt"

# инициализируем клиент, с помощью которого будем подключаться к Redis
    client = redis.StrictRedis(
            host=host,
            port=port,
            password=password,
            ssl=True,
            ssl_ca_certs=ca_path)

# просто подключиться недостаточно, да и неинтересно: нужно же ещё и поработать с хранилищем


# получим значение ключа с помощью команды get
    result = client.get("626a81ce9a8cd1920641e264")
    if not result:
        print("Запись с указанным ключом не найдена.")
        return

    result = result.decode("utf-8")


    print(result)

if __name__ == '__main__':
    main()

{"_id": "626a81ce9a8cd1920641e264", "name": "Гаврила Архипович Игнатьев", "login": "gavaii_2222", "update_ts_utc": "2023-04-15 08:18:00"}
